doc/wcmp/wcmp-design.md

```diff
@@ -0,0 +1,605 @@
+in order to expose the default state.
+
+**Skeleton code:**
+```jinja
+{
+    ...
+
+    "BGP_DEVICE_GLOBAL": {
+        "STATE": {
+            "wcmp_enabled": "false"
+        }
+    },
+
+    ...
+}
+```
+
+## 2.5 Flows
+
+### 2.5.1 Config section
+
+### 2.5.1.1 WCMP update
+
+![WCMP update flow](images/wcmp_update_flow.svg "Figure 3: WCMP update flow")
+
+###### Figure 3: WCMP update flow
+
+### 2.5.2 Show section
+
+#### 2.5.2.1 WCMP show
+
+![WCMP show flow](images/wcmp_show_flow.svg "Figure 4: WCMP show flow")
+
+###### Figure 4: WCMP show flow
+
+## 2.6 CLI
+
+### 2.6.1 Command structure
+
+**User interface**:
+```
+config
+|--- bgp
+     |--- device-global
+          |--- wcmp <ARG>
+
+show
+|--- bgp
+     |--- device-global OPTIONS
+```
+
+**Arguments:**
+
+_config bgp device-global wcmp_
+1. `enabled|disabled` - enable/disable WCMP using BGP link bandwidth
+
+**Options:**
+
+_show bgp device-global_
+1. `-j|--json` - display in JSON format
+
+### 2.6.2 Usage examples
+
+#### 2.6.2.1 Config command group
+
+**The following command updates WCMP configuration:**
+```bash
+config bgp device-global wcmp enabled
+```
+
+#### 2.6.2.2 Show command group
+
+**The following command shows WCMP configuration:**
+```bash
+show bgp device-global
+TSA        WCMP
+---------  --------
+Disabled   Enabled
+
+show bgp device-global --json
+{
+    "tsa": "disabled",
+    "wcmp": "enabled"
+}
+```
+
+## 2.7 YANG model
+
+Existing YANG model `sonic-bgp-device-global.yang` at `sonic-buildimage/src/sonic-yang-models/yang-models`  
+will be extended with a new schema in order to provide support for WCMP.
+
+**Skeleton code:**
+```yang
+module sonic-bgp-device-global {
+    yang-version 1.1;
+
+    namespace "http://github.com/sonic-net/sonic-bgp-device-global";
+    prefix bgp_device_global;
+
+    organization "SONiC";
+    contact "SONiC";
+    description "SONIC Device-specific BGP global data";
+
+    revision 2024-01-28 {
+        description "Add Weighted ECMP using BGP link bandwidth";
+    }
+
+    revision 2022-06-26 {
+        description "Initial revision";
+    }
+
+    container sonic-bgp-device-global {
+
+        container BGP_DEVICE_GLOBAL {
+
+            container STATE {
+
+                ...
+
+                leaf wcmp_enabled {
+                    description "Enables/Disables Weighted-Equal Cost-Multipath using BGP link bandwidth";
+                    type boolean;
+                    default "false";
+                }
+
+            }
+            /* end of STATE container */
+        }
+        /* end of BGP_DEVICE_GLOBAL container */
+    }
+    /* end of sonic-bgp-device-global container */
+}
+/* end of module sonic-bgp-device-global */
+```
+
+## 2.8 Warm/Fast boot
+
+No special handling is required
+
+# 3 Test plan
+
+## 3.1 Unit tests via PYTEST
+
+WCMP basic configuration test:
+1. Verify FRR route-map is generated properly after WCMP config update
+
```

Comment:
We may need a UT to verify whether the WCMP configs are applied after save and reboot.

Reply:
@venkatmahalingam simple UTs are not intended for such a testing

Reply:
Ok, can you point to the comprehensive list of test cases for this feature?

Reply:
@venkatmahalingam sure. I will update you once the test plan is ready